src/solver/take/forward_take.cpp

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/take/invoke_params.hpp>
+#include <miopen/take/solvers.hpp>
+#include <miopen/take.hpp>
+#include <miopen/target_properties.hpp>
+
+// TODO: as my function is in 2024H2, I don't have test examples now
+// I have tuned this param with random tests I generated
+#define LOCAL_SIZE 1024
+
+namespace miopen {
+
+namespace solver {
+
+namespace take {
+
+bool TakeForward::IsApplicable(const ExecutionContext& context,
+                               const miopen::take::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+        return false;
+    if(!problem.IsRightLength())
+        return false;
+    if(!problem.IsAllPacked())
+        return false;
+    return true;
+}
+
+ConvSolution TakeForward::GetSolution(const ExecutionContext& context,
+                                      const miopen::take::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+    auto ydims        = problem.GetYDesc().GetLengths();
+    auto output_numel =
```

Comment:
use yDesc.GetElementSize()

Reply:
Fixed it in [this commit](https://github.com/moreh-dev/MIOpen/pull/9/commits/49ae1dd03bce98973b224707262a5e7b4df3452a)